docs/contributing.md

```diff
@@ -81,58 +97,35 @@ If you're a MetaMask contributor, you can create these preview versions via draf
 
    - **Yarn >= 2 (berry):**
 
-     Add the following in `.yarnrc.yml`
+     Ensure that the project `.yarnrc.yml` file has the following contents:
 
      ```
-     npmScopes:
-        metamask:
-           npmAlwaysAuth: true
-           npmAuthToken: <your personal access token>
-           npmRegistryServer: 'https://npm.pkg.github.com'
-     ```
-
-   Make sure not to commit these changes.
-
-3. Go to GitHub and open up a pull request for this repository, then post a comment on the PR with the text `@metamaskbot publish-preview`. (This triggers the `publish-preview` GitHub action.)
-4. After a few minutes, you will see a new comment indicating that all packages have been published with the format `<package name>-<commit id>`.
-5. Switch back to your project locally and update `package.json` by replacing the versions for the packages you've changed in your PR using the new version format (e.g. `1.2.3-e2df9b4` instead of `~1.2.3`), then run `yarn install`.
-6. Repeat steps 3-5 after pushing new changes to your PR to generate and use new preview versions.
-
-### As an independent contributor
+     npmRegistries:
+       "https://npm.pkg.github.com":
+         npmAlwaysAuth: true
+         npmAuthToken: "${GITHUB_NPM_TOKEN-}"
 
-If you're a contributor and you've forked this repository, you can create preview versions for a branch via provided scripts:
-
-1. Navigate to your settings within GitHub and [create a **classic** access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `read:packages` scope.
-2. Switch to your project locally and add/edit the appropriate file with the following content, filling in the appropriate areas.
-
-   - **Yarn 1 (classic) or NPM:**
-
-     Add the following in `.npmrc`
-
-     ```
-     @<your GitHub username>:registry=https://npm.pkg.github.com
-     //npm.pkg.github.com/:_authToken=<your personal access token>
+     npmScopes:
+       metamask:
+         npmRegistryServer: "${METAMASK_NPM_REGISTRY:-https://registry.yarnpkg.com}"
      ```
 
-   - **Yarn >= 2 (berry):**
+     The `METAMASK_NPM_REGISTRY` environment variable lets you control which registry is used for `@metamask`-scoped packages. Set this environment variable to `https://npm.pkg.github.com` to use preview builds. The `GITHUB_NPM_TOKEN` environment variable is where your token is set (the one created in step 1).
 
-     Add the following in `.yarnrc.yml`
+     For example, in Bash you can set both of these environment variables when installing dependencies:
```

Comment:
Nit: This syntax should work in any shell, so we could say:

``` suggestion
     For example, on the command line, you can set both of these environment variables when installing dependencies:
```

Reply:
Maybe ever POSIX shell? Not sure it would work on Windows (e.g. Powershell or cmd.exe)

Reply:
Oh yeah, good point.

Considering that ZSH is default on macOS and Bash is the most popular shell, I guess we could cover our bases in a clear way by saying:

``` suggestion
     For example, in Bash or ZSH, you can set both of these environment variables when installing dependencies:
```

Reply:
Good call